src/main/java/eu/sshoc/dataversesuperset/DataSaver.java

```diff
@@ -68,17 +74,44 @@ public String createTable(DataInfo dataInfo) throws IOException {
 				+ dataInfo.columns.stream().map(c -> "?").collect(Collectors.joining(", "))
 				+ ")";
 
-		List<Object[]> rows = dataInfo.reader.createDBInserts(dataInfo, dataLoader);
-
-		int batchLimit = 9000;
-		for(int i = batchLimit; i < rows.size(); i+=batchLimit) {
-			jdbcTemplate.batchUpdate(insertValues, rows.subList(0, i));
+		List<Object[]> rows = new ArrayList<>();
+		try (CloseableHttpResponse response = openFile(dataInfo)) {
+			int columnCount = dataInfo.columns.size();
+			Reader reader = Reader.createReader(dataInfo, response.getEntity());
+			int batchLimit = 9000;
+			while (reader.hasNext()) {
+				String[] record = reader.next();
+				Object[] row = new Object[columnCount];
+				for (int i = 0; i < columnCount; i++) {
+					DataInfo.ColumnType type = dataInfo.columns.get(i).type;
+					row[i] = DataInfo.VALUE_PARSERS.get(type).parse(record[i]);
+				}
+				rows.add(row);
+				if (rows.size() % batchLimit == 0 || !reader.hasNext()) {
+					jdbcTemplate.batchUpdate(insertValues, rows);
+					rows = new ArrayList<>();
+				}
+			}
 		}
-		jdbcTemplate.batchUpdate(insertValues, rows);
 
 		return tableName;
 	}
-	
+
+	protected CloseableHttpResponse openFile(DataInfo dataInfo) throws IOException {
+		HttpGet httpGet = new HttpGet(dataInfo.fileUrl);
+		CloseableHttpResponse response = HttpClients.createDefault().execute(httpGet);
+		try {
+			int statusCode = response.getStatusLine().getStatusCode();
+			HttpEntity entity = response.getEntity();
+			if (statusCode != HttpStatus.OK.value() || entity == null || entity.getContentType() == null)
+				throw new IOException("status code: " + statusCode);
```

Comment:
This will result in error 500, but it's more of an external problem.
See logger and exception in `DataLoader`. Although now I think `NOT_FOUND` is not a great status either (it suggests the dataverse-superset service doesn't exist), you can change it to `BAD_REQUEST`.